src/main/java/org/subethamail/smtp/server/Session.java

```diff
@@ -196,56 +196,68 @@ private void runCommandLoop() throws IOException {
             return;
         }
 
-        this.sendResponse(
-                "220 " + this.server.getHostName() + " ESMTP " + this.server.getSoftwareName());
+        final SessionAcceptance sresult = this.server.getSessionHandler().accept(this);
+        if (!sresult.isAccepted()) {
+            log.debug("SMTP " + sresult.getErrorMessage());
+            this.sendResponse(sresult.getErrorCode() + " " + sresult.getErrorMessage());
+            return;
+        }
 
-        while (!this.quitting) {
-            try {
-                String line = null;
-                try {
-                    line = this.reader.readLine();
-                } catch (SocketException ex) {
-                    // Lots of clients just "hang up" rather than issuing QUIT,
-                    // which would
-                    // fill our logs with the warning in the outer catch.
-                    if (log.isDebugEnabled())
-                        log.debug("Error reading client command: " + ex.getMessage(), ex);
+        try {
+            this.sendResponse(
+                    "220 " + this.server.getHostName() + " ESMTP " + this.server.getSoftwareName());
 
+            while (!this.quitting) {
```

Comment:
What is all this? Formatting? Can you not change this please.

Reply:
This is not "totally" a formatting. I had to add "this.server.getSessionHandler().onSessionEnd(this);" on a finally clause to be sure to be invoked. Given the new try {} finally {} the original code is more indented.
I could revert indenting to have a cleaner PR but then the code wouldn't so much clean